README.md

```diff
@@ -201,6 +201,38 @@ Note that `#count` is calculated automatically based on the number of batches in
 your collection, and your Task’s progress will be displayed in terms of batches
 (not the number of records in the relation).
 
+On databases with large tables, sometime using an `ActiveRecord::Relation` to define a collection will
+cause your queries to timeout. One pattern to to have performant tasks is to look up the range of IDs that
+need to be operated on, then create a collection of ranges to query your models. For example:
+
+```ruby
+module Maintenance
+  class BackfillALotOfPosts < MaintenanceTasks::Task
+    # Generates an array like
+    # [(1..1_000_000),(1_000_001..2_000_000),(2_000_001..3_000_000), ...]
+    def collection
+      ranges = []
+      100.times do |i|
+        range_start = (1_000_000 * i) + 1
+        range_end = 1_000_000 * (i + 1)
+        ranges << (range_start..range_end)
+      end
+      ranges
+    end
+
+    def process(range)
+      Post
+        .where(id: range, some_column: "Some Value")
+        .in_batches { |batch| batch.update_all(some_column: "Some New Value") }
+    end
+  end
+end
+```
+
+This pattern will keep your individual queries fast, while performing updates in small batches.
+With this approach, updating or deleteing 10s of millions of records can take under an hour, even if
+the columns being accessed/changed are not indexed.
```

Comment:
>are not indexed.

I'm a little concerned about this claim since a relation like `where(id: 1_000_000...2_000_000, some_column: "Some Value")` should generate SQL like `where id=>1000000 and id<=2000000 and some_column="Some Value"`  and if there are plenty of records in the `id` range then database will have to apply the `some_column` in-memory

Which is not drastically different from using `Post.where(some_column: "Some Value").in_batches` as a `collection`

Reply:
>Which is not drastically different from using `Post.where(some_column: "Some Value").in_batches` as a collection

I may be misunderstanding your point, but the difference is that when you use `Post.where(some_column: "Some Value").in_batches` in the collection, it will do a full table scan for `some_column` to build the association before batching it out.

I also thought that it would be the same as `in_batches`, but in practice, there were queries that resulted in full table scans, whereas this approach only does a scan on the ID that is the size of the range...and you can make the range as small or as large as necessary for the operation you're performing.

In our case, 1M records was fast and easy to grok, though 2-3M records could have worked in our case. In other cases, a much smaller range may be better.